src/Domain/Insights/SyntaxCheck.php

```diff
@@ -113,12 +112,20 @@ private function getTarget(): string
         return '.';
     }
 
-    private function vendorParentPathFind(): string
+    /**
+     * Recursively search for composer binary folder path
+     */
+    private function composerBinaryFolderFind(string $directory): string
     {
-        $baseProjectPath = \dirname(__DIR__, 3);
-        if (file_exists($baseProjectPath . DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR . 'autoload.php')) {
-            return $baseProjectPath;
+        $composerBinaryFolder = DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR . 'bin';
```

Comment:
`vendor/bin` shouldn't be hardcoded, it should respect composer's `config.bin-dir` setting.

Reply:
https://github.com/nunomaduro/phpinsights/pull/505/files#R44
Maybe use `composer exec -- <bin>` here to ensure parallel-lint is in the $PATH and let composer handle finding the bin dir?